### DISCUSSION TITLE
RFC - Rename "master" branch and set to default

### DISCUSSION THREAD

**Discussion by RaddishIoW** [HaylLtd/cordova-plugin-background-geolocation#40] [General]
# Proposal
It is proposed to rename the `master` branch to `stable`, and set this branch as the default.
When a release is created from the `develop` branch, that commit should be tagged and also merged into the `stable` branch.
This would provide a stable view of the project as it is for the latest public release, including documentation, which would be the default landing area for developers/users coming to the repository for information.

The workflow for merging the changes into the proposed `stable` branch should be integrated into the automatic process for creating a release - i.e. GitHub Actions.

**Comment by HarelM** [comment]
Seems to me like a overhead...
I prefer to keep it a bit mire "agile" - since development is done in breaches the extra step of merging to another branch seems redundant to me.
If the documentation have a version like you suggested than the only questing is which documentation to present by default, latest from code or latest released version.
In any case, I don't have strong feelings about it, if it works for you I'm fine with it...

**Reply by RaddishIoW** [reply]
If the step of merging from the `develop` branch to the `stable` branch is automated when creating a release (like we do with uploading to NPM), then there's nothing manual to do. It does mean that there's a stable view of the plugin for users when they land on the project on GitHub...
Basically, it's the latest public release, while active development happens on the `develop` branch by merging PRs like you say. Makes it clear for users that the code and docs you're looking at are not including changes that aren't in what you're using.

**Comment by RaddishIoW** [comment]
I've made this change, and created a PR for a workflow to automate merging when creating a release.

**Reply by HarelM** [reply]
I just released 2.0.
The extra branch was a huge pain.
I'm pretty sure the changelog automation is now broken.
I changed it to only update the changelog when commiting to stable to avoid a case where the changelog file will create a conflict that won't allow the develop branch to be merge to stable.
I hope I fixed most of the problem, but in general, as I said, seems like an overkill for this project...
